File: stepik-union/src/main/java/org/stepik/plugin/actions/step/CodeQuizAction.java
package org.stepik.plugin.actions.step;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.Presentation;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.stepik.core.courseFormat.StepNode;

import javax.swing.*;

import static org.stepik.core.courseFormat.StepType.CODE;

abstract class CodeQuizAction extends AbstractStepAction {
    CodeQuizAction(
            @Nullable String text,
            @Nullable String description,
            @Nullable Icon icon) {
        super(text, description, icon);
    }

    @Nullable
    @Contract("null -> null")
    static StepNode getCurrentCodeStepNode(@Nullable Project project) {
        StepNode stepNode = getCurrentStep(project);
        if (stepNode == null || stepNode.getType() != CODE) {
            return null;
        }
        return stepNode;
    }

    @Override
    public void update(@NotNull AnActionEvent e) {
        super.update(e);
        Presentation presentation = e.getPresentation();

        if (!presentation.isEnabled()) {
            return;
        }

        StepNode stepNode = getCurrentStep(e.getProject());
        presentation.setEnabled((stepNode != null) && (stepNode.getType() == CODE));
    }
}

File: stepik-union/src/main/kotlin/org/stepik/core/actions/step/CodeQuizAction.kt
package org.stepik.core.actions.step

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.project.Project
import org.jetbrains.annotations.Contract
import org.stepik.core.courseFormat.StepNode
import org.stepik.core.courseFormat.StepType.CODE
import javax.swing.Icon

abstract class CodeQuizAction protected constructor(
        text: String?,
        description: String?,
        icon: Icon?) : AbstractStepAction(text, description, icon) {

    override fun update(e: AnActionEvent?) {
        super.update(e)
        val presentation = e?.presentation

        if (presentation?.isEnabled != true) {
            return
        }

        presentation.isEnabled = getCurrentStep(e.project)?.type == CODE
    }

    companion object {
        @Contract("null -> null")
        fun getCurrentCodeStepNode(project: Project?): StepNode? {
            val stepNode = getCurrentStep(project)
            return if (stepNode?.type == CODE) stepNode else null
        }
    }
}
